agent/stats/utils_unix.go

```diff
@@ -39,9 +39,8 @@ func dockerStatsToContainerStats(dockerStats *types.StatsJSON) (*ContainerStats,
 }
 
 func validateDockerStats(dockerStats *types.StatsJSON) error {
-	// The length of PercpuUsage represents the number of cores in an instance.
-	if len(dockerStats.CPUStats.CPUUsage.PercpuUsage) == 0 || numCores == uint64(0) {
```

Comment:
So, we think that `len(dockerStats.CPUStats.CPUUsage.PercpuUsage) == 0` was redundant or a mistake? I'm hesitant to remove something we don't seem to have clarity on why it was introduced.

Reply:
PercpuUsage doesn't appear to be used anywhere in agent, and it's not available anymore with cgroupv2. I'm not sure how else we can check that it's safe to remove? FWIW there isn't really any alternative to just removing the check from what I can tell, since cgroupv2 stats simply dont provide this level of granularity.

If you would prefer we could just have an if-statement here and continue checking this array when we're using cgroupv1?

Reply:
I'm guessing the intention was to do a simple sanity check of the stats API.
There is an alternative to the `PercpuUsage` field, `online_cpus`, according to docker API [doc](https://docs.docker.com/engine/api/v1.41/#operation/ContainerStats), which should be available when either v1 or v2 cgroups is used:
```
number_cpus = lenght(cpu_stats.cpu_usage.percpu_usage) or cpu_stats.online_cpus
```
But then I also saw this comment
```
If either precpu_stats.online_cpus or cpu_stats.online_cpus is nil then for compatibility with older daemons the length of the corresponding cpu_usage.percpu_usage array should be used.
```
Earliest docker API reference I could find is for [1.18](https://docs.docker.com/engine/api/v1.18/) and it does not contain the `online_cpu` field. 

Since ECSAgent-supported docker api version can go back to [1.17](https://github.com/aws/amazon-ecs-agent/blob/225bc3a556bd2d1759ab27b23f54e7e68086c9f0/agent/dockerclient/versions.go#L19), which can still work with 20.10 docker engine (based on this [doc](https://docs.docker.com/engine/api/)) which is when docker introduced cgroupv2, it's not 100% safe for us to check `online_cpus` when cgroupv2 is used. 

We can still keep the v1 check, but I'm not sure of a safe way to validate docker stats numCPU when using cgroupv2.